### PR TITLE
Add support for the focal Tversky loss

### PR DESCRIPTION
**Pull request by zifuwanggg**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

The focal Tversky loss was proposed in https://arxiv.org/abs/1810.07842. It has nearly 600 citations and has been shown to be extremely useful for highly imbalanced (medical) datasets. To add support for the focal Tversky loss, only few lines of changes are needed for the Tversky loss.

## Modification

Add `gamma` as (optional) argument in the constructor of `TverskyLoss`. This parameter is then passed to `tversky_loss` to compute the focal Tversky loss.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zifuwanggg** [comment]
It seems that I have signed CLA but still get complains. 

Screen Shot 2023-03-22 at 22 26 13
Screen Shot 2023-03-22 at 22 26 26

**Comment by xiexinch** [comment]
Hi @zifuwanggg,
Thanks for your contribution!
You can check if the email used in the local commit is the same as the email in the GitHub profile, if not, please add the email used in the commit to the GitHub profile.

File: mmseg/models/losses/tversky_loss.py [hunk 1]
```diff
@@ -75,6 +78,9 @@ class TverskyLoss(nn.Module):
         beta (float, in [0, 1]):
             The coefficient of false negatives. Default: 0.7.
             Note: alpha + beta = 1.
+        gamma (float, in [1, 3]): The focal term. When `gamma` > 1, the loss
```
Comment (MengzhangLI) [inline mmseg/models/losses/tversky_loss.py:81]: We may add assertion below to ensure `1.0 < gamma < 3.0`, so do unit test.

File: mmseg/models/losses/tversky_loss.py [hunk 2]
```diff
@@ -31,6 +32,8 @@ def tversky_loss(pred,
                 alpha=alpha,
                 beta=beta,
                 smooth=smooth)
+            if gamma > 1:
+                tversky_loss **= (1 / gamma)
```
Comment (xiexinch) [inline mmseg/models/losses/tversky_loss.py:36]: Might add a unit test.

File: mmseg/models/losses/tversky_loss.py (hunk 1 above)
Comment (zifuwanggg) [reply mmseg/models/losses/tversky_loss.py:81]: The [1, 3] range was mentioned in the paper, although they do not mention why `gamma` should be in this range. Theoretically, `gamma` can be any real number. Practically, a small `gamma` (e.g. < 2) works better. Perhaps we should remove the requirement for `gamma` in [1, 3] and let the user to decide?

Comment (zifuwanggg) [reply mmseg/models/losses/tversky_loss.py:81]: Beside, I also think `alpha` and `beta` in the original `TverskyLoss` class does not need to satisfy `assert alpha + beta == 1.0`. Floating point equality is dangerous and according the definition of the [Tversky index](https://en.wikipedia.org/wiki/Tversky_index), the only requirement for `alpha` and `beta` is that they are non-negative numbers.

**Comment by zifuwanggg** [comment]
> 

Thanks for the comment. The email I used for the very first commits was indeed by default `username@localhost.local`. Even if I add that email to my GitHub account, I cannot verify that email address since it cannot receive the verification email from GitHub. Modify the email address for previous commits are strongly [discouraged](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user). Is there a workaround to let me sign the CLA, or I might just close this one and open a new pull request?

**Comment by xiexinch** [comment]
> Hi, @xiexinch
> 
> Thanks for the comment. The email I used for the very first commits was indeed by default `username@localhost.local`. Even if I add that email to my GitHub account, I cannot verify that email address since it cannot receive the verification email from GitHub. Modify the email address for previous commits are strongly [discouraged](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user). Is there a workaround to let me sign the CLA, or I might just close this one and open a new pull request?

Hi @zifuwanggg,
In that case, I think it's better to close this pull request and create a new one :)

**Comment by OpenMMLab-Assistant-007** [comment]
Hi
@zifuwanggg 
We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤
